raspberry-pi/4/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, pkgs, ...}:
+
+{
+  boot = {
+    kernelPackages = lib.mkDefault pkgs.linuxPackages_rpi4;
+    initrd.availableKernelModules = [ "usbhid" "usb_storage" "vc4" ];
```

Comment:
Not sure about `vc4` but are the other modules not also included in `nixos-generate-config` output?

Reply:
I'd like for this file to be a substitute for `nixos-generate-config` so that there's one less step when installing NixOS.

cc @samueldr for `vc4`.

Reply:
I've searched google and nixpkgs but couldn't find reference for that module, what hardware bit is it used for?

Reply:
For the mainline kernel it's actually `pcie-brcmstb`/`pcie_brcmstb` (interchangeable with our tooling):

 - https://github.com/NixOS/nixpkgs/blob/65670311112f09f8e36b5ec714de0c222e61a09f/nixos/modules/profiles/all-hardware.nix#L81

Sorry I had the name wrong.

Though it's highly possible it't built as a built-in module for the vendor kernel.

Reply:
Seems to be in already in the kernel.